Zend/tests/033.phpt

```diff
@@ -62,5 +64,16 @@ Warning: Trying to access array offset on value of type null in %s on line %d
 Warning: Trying to access array offset on value of type null in %s on line %d
 
 Warning: Trying to get property 'foo' of non-object in %s on line %d
+
+Warning: Undefined variable: arr in %s on line %d
```

Comment:
It's going to be hard to explain why `$a[2]->x = 2;` warns about undefined variable but `$a[2][2] = 2;` does not.

Reply:
It may be hard to explain why arrays and objects have different semantics in the first place, but given that they do, this seems more consistent:

`$a[0] = 42` where `$a` is undefined is considered legal and does not throw any notices or errors. `$a->x = 42` is considered illegal and throws a warning in PHP 7 and exception in PHP 8. In this case, the undefined variable notice indicates the "root cause" that needs to be fixed, while the "Attempt to assign property 'b' of non-object" is a followup error caused by it.